### PR TITLE
Moved all DOM helpers in src/node.js

### PR DESCRIPTION
**Pull request by regiskuckaertz**
In reference to https://github.com/guardian/scribe/issues/397

This is a first step, I spotted a lot of redundant code that could be replaced with calls to DOM helpers. I'll leave that to another PR not to overwhelm you guys.

### CODE REVIEW

File: src/node.js [hunk 1]
```diff
@@ -34,6 +67,35 @@ define(function () {
...
+      currentNode = currentNode.parentNode;
+    }
+  }
+
+  function nextAll(node) {
+    var all = [];
+    while (node = node.nextSibling) {
+      all.push(node);
+    }
+    return all;
```
Comment (rrees) [inline src/node.js]: Can we make this return an immutable collection?

File: src/node.js [hunk 2]
```diff
@@ -34,6 +67,35 @@ define(function () {
...
+    // instance, so stop traversing there.
+    while (currentNode && ! isTopContainerElement(currentNode)) {
+      if (nodeFilter(currentNode)) {
+        return currentNode;
+      }
+      currentNode = currentNode.parentNode;
+    }
+  }
+
+  function nextAll(node) {
```
Comment (rrees) [inline src/node.js]: Maybe call this allFollowingNodes? The name isn't quite right.

**Comment by rrees** [comment]
Looks like a sensible consolidation :+1: 

I wouldn't want to see files of more than 200 or 300 lines though.

**Comment by regiskuckaertz** [comment]
I agree 100%, at some point this module can be split in sub-modules, similar to api.js.

File: src/node.js (hunk 2 above)
Comment (regiskuckaertz) [inline src/node.js]: Sure!

File: src/node.js (hunk 1 above)
Comment (regiskuckaertz) [inline src/node.js]: Oh, that's a good idea, I'm on it.

File: src/node.js (hunk 2 above)
Comment (regiskuckaertz) [inline src/node.js]: What do you think of nextSiblings? More idiomatic, though using inflections might throw people off ("I did not see the s").

Comment (rrees) [inline src/node.js]: `nextSiblings` sounds a good choice for now

**Comment by rrees** [comment]
Can you merge in the upstream master? Then I'll try and release next week.

**Pull request merged**